### ISSUE TITLE
FTBFS on SLES15, if.h namespace collision?

### ISSUE THREAD

**Issue by thresheek** [acassen/keepalived#1079]
When building current git (and 2.0.10 fwiw, didnt test earlier versions) on SLES15, the following error is seen:
```
thresh@sles15-ka:~/keepalived> export LC_ALL=C && git describe && autoreconf -fisv && ./configure --disable-nftables && make -j1
v1.2.2-3556-gdcefd33a
autoreconf: Entering directory `.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force 
autoreconf: configure.ac: tracing
autoreconf: configure.ac: not using Libtool
autoreconf: running: /usr/bin/autoconf --force
autoreconf: running: /usr/bin/autoheader --force
autoreconf: running: automake --add-missing --force-missing
configure.ac:288: installing './ar-lib'
configure.ac:278: installing './compile'
configure.ac:98: installing './install-sh'
configure.ac:98: installing './missing'
genhash/Makefile.am: installing './depcomp'
autoreconf: Leaving directory `.'
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether make supports nested variables... (cached) yes
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking for style of include used by make... GNU
checking dependency style of gcc... gcc3
checking whether make sets $(MAKE)... (cached) yes
checking for ranlib... ranlib
checking for grep that handles long lines and -e... /usr/bin/grep
checking whether ln -s works... yes
checking for a sed that does not truncate output... /usr/bin/sed
checking for strip... strip
checking for ldd... ldd
checking for ar... ar
checking the archiver (ar) interface... ar
checking for -Wimplicit-fallthrough... yes
checking for PIE support... yes
checking for -Wformat -Werror=format-security support... yes
checking for -Wp,-D_FORTIFY_SOURCE=2 support... yes
checking for -fexceptions support... yes
checking for -fstack-protector-strong support... yes
checking for --param=ssp-buffer-size=4 support... yes
checking for -grecord-gcc-switches support... yes
checking for -Wl,-z,relro support... yes
checking for -Wl,-z,now support... yes
checking for -O2 support... yes
checking for clock_gettime() requires -lrt... no
checking how to run the C preprocessor... gcc -E
checking for egrep... /usr/bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking arpa/inet.h usability... yes
checking arpa/inet.h presence... yes
checking for arpa/inet.h... yes
checking fcntl.h usability... yes
checking fcntl.h presence... yes
checking for fcntl.h... yes
checking limits.h usability... yes
checking limits.h presence... yes
checking for limits.h... yes
checking netdb.h usability... yes
checking netdb.h presence... yes
checking for netdb.h... yes
checking netinet/in.h usability... yes
checking netinet/in.h presence... yes
checking for netinet/in.h... yes
checking for stdint.h... (cached) yes
checking for stdlib.h... (cached) yes
checking for string.h... (cached) yes
checking sys/ioctl.h usability... yes
checking sys/ioctl.h presence... yes
checking for sys/ioctl.h... yes
checking sys/param.h usability... yes
checking sys/param.h presence... yes
checking for sys/param.h... yes
checking sys/prctl.h usability... yes
checking sys/prctl.h presence... yes
checking for sys/prctl.h... yes
checking sys/socket.h usability... yes
checking sys/socket.h presence... yes
checking for sys/socket.h... yes
checking sys/time.h usability... yes
checking sys/time.h presence... yes
checking for sys/time.h... yes
checking syslog.h usability... yes
checking syslog.h presence... yes
checking for syslog.h... yes
checking for unistd.h... (cached) yes
checking asm/types.h usability... yes
checking asm/types.h presence... yes
checking for asm/types.h... yes
checking linux/ethtool.h usability... yes
checking linux/ethtool.h presence... yes
checking for linux/ethtool.h... yes
checking linux/icmpv6.h usability... yes
checking linux/icmpv6.h presence... yes
checking for linux/icmpv6.h... yes
checking linux/if_ether.h usability... yes
checking linux/if_ether.h presence... yes
checking for linux/if_ether.h... yes
checking linux/if_packet.h usability... yes
checking linux/if_packet.h presence... yes
checking for linux/if_packet.h... yes
checking linux/ip.h usability... yes
checking linux/ip.h presence... yes
checking for linux/ip.h... yes
checking linux/sockios.h usability... yes
checking linux/sockios.h presence... yes
checking for linux/sockios.h... yes
checking linux/types.h usability... yes
checking linux/types.h presence... yes
checking for linux/types.h... yes
checking for linux/fib_rules.h... yes
checking for linux/if_addr.h... yes
checking for linux/if_link.h... yes
checking for linux/if_arp.h... yes
checking for stdbool.h that conforms to C99... yes
checking for _Bool... yes
checking for inline... inline
checking for int64_t... yes
checking for pid_t... yes
checking for size_t... yes
checking for uint16_t... yes
checking for uint32_t... yes
checking for uint64_t... yes
checking for uint8_t... yes
checking for an ANSI C-conforming const... yes
checking vfork.h usability... no
checking vfork.h presence... no
checking for vfork.h... no
checking for fork... yes
checking for vfork... yes
checking for working fork... yes
checking for working vfork... (cached) yes
checking for stdlib.h... (cached) yes
checking for GNU libc compatible malloc... yes
checking for stdlib.h... (cached) yes
checking for GNU libc compatible realloc... yes
checking for dup2... yes
checking for getcwd... yes
checking for gettimeofday... yes
checking for memmove... yes
checking for memset... yes
checking for select... yes
checking for setenv... yes
checking for socket... yes
checking for strcasecmp... yes
checking for strchr... yes
checking for strdup... yes
checking for strerror... yes
checking for strpbrk... yes
checking for strstr... yes
checking for strtol... yes
checking for strtoul... yes
checking for uname... yes
checking for pipe2... yes
checking for signalfd... yes
checking for inotify_init1... yes
checking for vsyslog... yes
checking for epoll_create1... yes
checking whether ETHERTYPE_IPV6 is declared... yes
checking openssl/ssl.h usability... yes
checking openssl/ssl.h presence... yes
checking for openssl/ssl.h... yes
checking openssl/err.h usability... yes
checking openssl/err.h presence... yes
checking for openssl/err.h... yes
checking openssl/md5.h usability... yes
checking openssl/md5.h presence... yes
checking for openssl/md5.h... yes
checking for MD5_Init in -lcrypto... yes
checking for SSL_CTX_new in -lssl... yes
checking SSL_set_tlsext_host_name() - may be a definition... yes
checking for SSL_CTX_set_verify_depth... yes
checking for SSL_set0_rbio... yes
checking for OPENSSL_init_crypto... yes
checking for TLS_method... yes
checking whether IPV4_DEVCONF_ARP_IGNORE is declared... yes
checking whether IPV4_DEVCONF_ACCEPT_LOCAL is declared... yes
checking whether IPV4_DEVCONF_RP_FILTER is declared... yes
checking whether IPV4_DEVCONF_ARPFILTER is declared... yes
checking for linux/rtnetlink.h... yes
checking for nl_socket_modify_cb in -lnl... no
configure: WARNING: keepalived will be built without libnl support.
checking for magic_open in -lmagic... no
checking whether RTA_ENCAP is declared... yes
checking whether RTA_EXPIRES is declared... yes
checking whether RTA_NEWDST is declared... yes
checking whether RTA_PREF is declared... yes
checking whether FRA_SUPPRESS_PREFIXLEN is declared... yes
checking whether FRA_SUPPRESS_IFGROUP is declared... yes
checking whether FRA_TUN_ID is declared... yes
checking whether RTAX_CC_ALGO is declared... yes
checking whether RTAX_QUICKACK is declared... yes
checking whether RTEXT_FILTER_SKIP_STATS is declared... yes
checking whether FRA_L3MDEV is declared... yes
checking whether FRA_UID_RANGE is declared... yes
checking whether RTAX_FASTOPEN_NO_COOKIE is declared... yes
checking whether RTA_VIA is declared... yes
checking whether FRA_OIFNAME is declared... yes
checking whether FRA_PROTOCOL is declared... no
checking whether FRA_IP_PROTO is declared... no
checking whether FRA_SPORT_RANGE is declared... no
checking whether FRA_DPORT_RANGE is declared... no
checking whether RTA_TTL_PROPAGATE is declared... yes
checking whether IFA_FLAGS is declared... yes
checking whether IP_MULTICAST_ALL is declared... yes
checking whether LWTUNNEL_ENCAP_MPLS is declared... yes
checking whether LWTUNNEL_ENCAP_ILA is declared... yes
checking linux/netfilter/x_tables.h usability... yes
checking linux/netfilter/x_tables.h presence... yes
checking for linux/netfilter/x_tables.h... yes
checking xtables.h usability... no
checking xtables.h presence... no
checking for xtables.h... no
checking for linux/if.h and net/if.h namespace collision... no
checking for linux/if_ether.h then netinet/if_ether.h namespace collision... no
checking for libiptc/libiptc.h linux/if.h and net/if.h namespace collision... yes
checking linux/ip_vs.h usability... yes
checking linux/ip_vs.h presence... yes
checking for linux/ip_vs.h... yes
checking whether IP_VS_SVC_F_ONEPACKET is declared... yes
checking whether IPVS_DEST_ATTR_ADDR_FAMILY is declared... yes
checking whether IPVS_DAEMON_ATTR_SYNC_MAXLEN is declared... yes
checking whether IPVS_DAEMON_ATTR_MCAST_GROUP is declared... yes
checking whether IPVS_DAEMON_ATTR_MCAST_GROUP6 is declared... yes
checking whether IPVS_DAEMON_ATTR_MCAST_PORT is declared... yes
checking whether IPVS_DAEMON_ATTR_MCAST_TTL is declared... yes
checking whether IPVS_SVC_ATTR_STATS64 is declared... yes
checking whether IPVS_DEST_ATTR_STATS64 is declared... yes
checking whether IFLA_MACVLAN_MODE is declared... yes
checking whether MACVLAN_MODE_PRIVATE is declared... yes
checking whether SOCK_NONBLOCK is declared... yes
checking whether SOCK_CLOEXEC is declared... yes
checking whether IPVS_SVC_ATTR_PE_NAME is declared... yes
checking whether O_PATH is declared... yes
checking whether GLOB_BRACE is declared... yes
checking whether FRA_SRC is declared... yes
checking whether IFLA_INET6_ADDR_GEN_MODE is declared... yes
checking whether IFLA_VRF_MAX is declared... yes
checking whether SO_MARK is declared... yes
checking whether CLONE_NEWNET is declared... yes
checking for setns... yes
checking whether SCHED_RR is declared... yes
checking whether RLIMIT_RTTIME is declared... yes
checking whether SCHED_RESET_ON_FORK is declared... yes
checking for sphinx-build... No
checking for rpm... Yes
checking for rpmbuild... Yes

checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating keepalived/Makefile
config.status: creating lib/Makefile
config.status: creating keepalived/core/Makefile
config.status: creating keepalived.spec
config.status: creating genhash/Makefile
config.status: creating keepalived/check/Makefile
config.status: creating keepalived/vrrp/Makefile
config.status: creating keepalived/bfd/Makefile
config.status: creating doc/Makefile
config.status: creating bin_install/Makefile
config.status: creating keepalived/dbus/Makefile
config.status: creating keepalived/etc/Makefile
config.status: creating keepalived/etc/init/Makefile
config.status: creating keepalived/etc/init.d/Makefile
config.status: creating doc/man/man8/Makefile
config.status: creating lib/config.h
config.status: executing depfiles commands

Keepalived configuration
------------------------
Keepalived version       : 2.0.10
Compiler                 : gcc
Preprocessor flags       : 
Compiler flags           : -Wall -Wunused -Wstrict-prototypes -Wextra -Winit-self -g -D_GNU_SOURCE -Wimplicit-fallthrough=3 -fPIE -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -O2
Linker flags             :  -pie
Extra Lib                :  -lcrypto  -lssl 
Use IPVS Framework       : Yes
IPVS use libnl           : No
IPVS syncd attributes    : Yes
IPVS 64 bit stats        : Yes
HTTP_GET regex support   : No
fwmark socket support    : Yes
Use VRRP Framework       : Yes
Use VRRP VMAC            : Yes
Use VRRP authentication  : Yes
With ip rules/routes     : Yes
Use BFD Framework        : No
SNMP vrrp support        : No
SNMP checker support     : No
SNMP RFCv2 support       : No
SNMP RFCv3 support       : No
DBUS support             : No
SHA1 support             : No
Use Json output          : No
libnl version            : None
Use IPv4 devconf         : Yes
Use iptables             : Yes
Use libiptc              : No
Use libipset             : No
Use nftables             : No
init type                : systemd
Strict config checks     : No
Build genhash            : Yes
Build documentation      : No

*** WARNING - this build will not support IPVS with IPv6. Please install libnl/libnl-3 dev libraries to support IPv6 with IPVS.

Making all in lib
make[1]: Entering directory '/home/thresh/keepalived/lib'
make  all-am
make[2]: Entering directory '/home/thresh/keepalived/lib'
  CC       memory.o
  CC       utils.o
  CC       notify.o
  CC       timer.o
  CC       scheduler.o
  CC       vector.o
  CC       list.o
  CC       html.o
  CC       parser.o
  CC       signals.o
  CC       logger.o
  CC       list_head.o
  CC       rbtree.o
  CC       process.o
  CC       rttables.o
  AR       liblib.a
make[2]: Leaving directory '/home/thresh/keepalived/lib'
make[1]: Leaving directory '/home/thresh/keepalived/lib'
Making all in keepalived
make[1]: Entering directory '/home/thresh/keepalived/keepalived'
Making all in core
make[2]: Entering directory '/home/thresh/keepalived/keepalived/core'
  CC       main.o
  CC       daemon.o
  CC       pidfile.o
  CC       layer4.o
  CC       smtp.o
  CC       global_data.o
  CC       global_parser.o
  CC       keepalived_netlink.o
  CC       namespaces.o
  AR       libcore.a
make[2]: Leaving directory '/home/thresh/keepalived/keepalived/core'
Making all in vrrp
make[2]: Entering directory '/home/thresh/keepalived/keepalived/vrrp'
  CC       vrrp_daemon.o
  CC       vrrp_print.o
  CC       vrrp_data.o
  CC       vrrp_parser.o
  CC       vrrp.o
In file included from ./../include/vrrp.h:32:0,
                 from ./../include/vrrp_arp.h:31,
                 from vrrp.c:45:
/usr/include/linux/if.h:82:2: error: redeclaration of enumerator 'IFF_UP'
  IFF_UP    = 1<<0,  /* sysfs */
  ^
/usr/include/net/if.h:44:5: note: previous definition of 'IFF_UP' was here
     IFF_UP = 0x1,  /* Interface is up.  */
     ^~~~~~
/usr/include/linux/if.h:83:2: error: redeclaration of enumerator 'IFF_BROADCAST'
  IFF_BROADCAST   = 1<<1,  /* __volatile__ */
  ^
/usr/include/net/if.h:46:5: note: previous definition of 'IFF_BROADCAST' was here
     IFF_BROADCAST = 0x2, /* Broadcast address valid.  */
     ^~~~~~~~~~~~~
/usr/include/linux/if.h:84:2: error: redeclaration of enumerator 'IFF_DEBUG'
  IFF_DEBUG   = 1<<2,  /* sysfs */
  ^
/usr/include/net/if.h:48:5: note: previous definition of 'IFF_DEBUG' was here
     IFF_DEBUG = 0x4,  /* Turn on debugging.  */
     ^~~~~~~~~
/usr/include/linux/if.h:85:2: error: redeclaration of enumerator 'IFF_LOOPBACK'
  IFF_LOOPBACK   = 1<<3,  /* __volatile__ */
  ^
/usr/include/net/if.h:50:5: note: previous definition of 'IFF_LOOPBACK' was here
     IFF_LOOPBACK = 0x8,  /* Is a loopback net.  */
     ^~~~~~~~~~~~
/usr/include/linux/if.h:86:2: error: redeclaration of enumerator 'IFF_POINTOPOINT'
  IFF_POINTOPOINT   = 1<<4,  /* __volatile__ */
  ^
/usr/include/net/if.h:52:5: note: previous definition of 'IFF_POINTOPOINT' was here
     IFF_POINTOPOINT = 0x10, /* Interface is point-to-point link.  */
     ^~~~~~~~~~~~~~~
/usr/include/linux/if.h:87:2: error: redeclaration of enumerator 'IFF_NOTRAILERS'
  IFF_NOTRAILERS   = 1<<5,  /* sysfs */
  ^
/usr/include/net/if.h:54:5: note: previous definition of 'IFF_NOTRAILERS' was here
     IFF_NOTRAILERS = 0x20, /* Avoid use of trailers.  */
     ^~~~~~~~~~~~~~
/usr/include/linux/if.h:88:2: error: redeclaration of enumerator 'IFF_RUNNING'
  IFF_RUNNING   = 1<<6,  /* __volatile__ */
  ^
/usr/include/net/if.h:56:5: note: previous definition of 'IFF_RUNNING' was here
     IFF_RUNNING = 0x40,  /* Resources allocated.  */
     ^~~~~~~~~~~
/usr/include/linux/if.h:89:2: error: redeclaration of enumerator 'IFF_NOARP'
  IFF_NOARP   = 1<<7,  /* sysfs */
  ^
/usr/include/net/if.h:58:5: note: previous definition of 'IFF_NOARP' was here
     IFF_NOARP = 0x80,  /* No address resolution protocol.  */
     ^~~~~~~~~
/usr/include/linux/if.h:90:2: error: redeclaration of enumerator 'IFF_PROMISC'
  IFF_PROMISC   = 1<<8,  /* sysfs */
  ^
/usr/include/net/if.h:60:5: note: previous definition of 'IFF_PROMISC' was here
     IFF_PROMISC = 0x100, /* Receive all packets.  */
     ^~~~~~~~~~~
/usr/include/linux/if.h:91:2: error: redeclaration of enumerator 'IFF_ALLMULTI'
  IFF_ALLMULTI   = 1<<9,  /* sysfs */
  ^
/usr/include/net/if.h:64:5: note: previous definition of 'IFF_ALLMULTI' was here
     IFF_ALLMULTI = 0x200, /* Receive all multicast packets.  */
     ^~~~~~~~~~~~
/usr/include/linux/if.h:92:2: error: redeclaration of enumerator 'IFF_MASTER'
  IFF_MASTER   = 1<<10, /* __volatile__ */
  ^
/usr/include/net/if.h:67:5: note: previous definition of 'IFF_MASTER' was here
     IFF_MASTER = 0x400,  /* Master of a load balancer.  */
     ^~~~~~~~~~
/usr/include/linux/if.h:93:2: error: redeclaration of enumerator 'IFF_SLAVE'
  IFF_SLAVE   = 1<<11, /* __volatile__ */
  ^
/usr/include/net/if.h:69:5: note: previous definition of 'IFF_SLAVE' was here
     IFF_SLAVE = 0x800,  /* Slave of a load balancer.  */
     ^~~~~~~~~
/usr/include/linux/if.h:94:2: error: redeclaration of enumerator 'IFF_MULTICAST'
  IFF_MULTICAST   = 1<<12, /* sysfs */
  ^
/usr/include/net/if.h:72:5: note: previous definition of 'IFF_MULTICAST' was here
     IFF_MULTICAST = 0x1000, /* Supports multicast.  */
     ^~~~~~~~~~~~~
/usr/include/linux/if.h:95:2: error: redeclaration of enumerator 'IFF_PORTSEL'
  IFF_PORTSEL   = 1<<13, /* sysfs */
  ^
/usr/include/net/if.h:75:5: note: previous definition of 'IFF_PORTSEL' was here
     IFF_PORTSEL = 0x2000, /* Can set media type.  */
     ^~~~~~~~~~~
/usr/include/linux/if.h:96:2: error: redeclaration of enumerator 'IFF_AUTOMEDIA'
  IFF_AUTOMEDIA   = 1<<14, /* sysfs */
  ^
/usr/include/net/if.h:77:5: note: previous definition of 'IFF_AUTOMEDIA' was here
     IFF_AUTOMEDIA = 0x4000, /* Auto media select active.  */
     ^~~~~~~~~~~~~
/usr/include/linux/if.h:97:2: error: redeclaration of enumerator 'IFF_DYNAMIC'
  IFF_DYNAMIC   = 1<<15, /* sysfs */
  ^
/usr/include/net/if.h:79:5: note: previous definition of 'IFF_DYNAMIC' was here
     IFF_DYNAMIC = 0x8000 /* Dialup device with changing addresses.  */
     ^~~~~~~~~~~
In file included from /usr/include/linux/netdevice.h:29:0,
                 from ./../include/vrrp_if.h:39,
                 from ./../include/vrrp.h:45,
                 from ./../include/vrrp_arp.h:31,
                 from vrrp.c:45:
/usr/include/linux/if.h:192:8: error: redefinition of 'struct ifmap'
 struct ifmap {
        ^~~~~
In file included from ./../include/vrrp.h:32:0,
                 from ./../include/vrrp_arp.h:31,
                 from vrrp.c:45:
/usr/include/net/if.h:111:8: note: originally defined here
 struct ifmap
        ^~~~~
In file included from /usr/include/linux/netdevice.h:29:0,
                 from ./../include/vrrp_if.h:39,
                 from ./../include/vrrp.h:45,
                 from ./../include/vrrp_arp.h:31,
                 from vrrp.c:45:
/usr/include/linux/if.h:229:8: error: redefinition of 'struct ifreq'
 struct ifreq {
        ^~~~~
In file included from ./../include/vrrp.h:32:0,
                 from ./../include/vrrp_arp.h:31,
                 from vrrp.c:45:
/usr/include/net/if.h:126:8: note: originally defined here
 struct ifreq
        ^~~~~
In file included from /usr/include/linux/netdevice.h:29:0,
                 from ./../include/vrrp_if.h:39,
                 from ./../include/vrrp.h:45,
                 from ./../include/vrrp_arp.h:31,
                 from vrrp.c:45:
/usr/include/linux/if.h:281:8: error: redefinition of 'struct ifconf'
 struct ifconf  {
        ^~~~~~
In file included from ./../include/vrrp.h:32:0,
                 from ./../include/vrrp_arp.h:31,
                 from vrrp.c:45:
/usr/include/net/if.h:176:8: note: originally defined here
 struct ifconf
        ^~~~~~
make[2]: *** [Makefile:422: vrrp.o] Error 1
make[2]: Leaving directory '/home/thresh/keepalived/keepalived/vrrp'
make[1]: *** [Makefile:583: all-recursive] Error 1
make[1]: Leaving directory '/home/thresh/keepalived/keepalived'
make: *** [Makefile:427: all-recursive] Error 1
thresh@sles15-ka:~/keepalived>
```

The following patch seems to workaround the issue.  I'm not submitting as a PR since I have no idea if it's right or not:
```
diff --git a/configure.ac b/configure.ac
index 7a2e8ce1..9badcc62 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1349,8 +1349,8 @@ AC_MSG_CHECKING([for linux/if.h and net/if.h namespace collision])
 SAV_CPPFLAGS="$CPPFLAGS"
 CPPFLAGS="$CPPFLAGS $kernelinc"
 AC_COMPILE_IFELSE([AC_LANG_SOURCE([[
-    #include <net/if.h>
     #include <linux/if.h>
+    #include <net/if.h>
   ]])],
   [
     AC_MSG_RESULT([no])
```

glibc-2.26-11.8.x86_64, kernel is kernel-default-4.12.14-23.1.x86_64

Let me know if you need more information.

**Comment by pqarmitage** [comment]
Can you try commenting out the `#include <net/if.h>` at line 31 of keepalived/include/vrrp.h and report back if that works.

**Comment by thresheek** [comment]
It does not help, same error with the following patch:
```
diff --git a/keepalived/include/vrrp.h b/keepalived/include/vrrp.h
index ae73a4b0..0781a9c8 100644
--- a/keepalived/include/vrrp.h
+++ b/keepalived/include/vrrp.h
@@ -29,7 +29,7 @@
 /* system include */
 #include <stdint.h>
 #include <stdbool.h>
-#include <net/if.h>
+/* ##include <net/if.h> */
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <arpa/inet.h>
```

**Comment by pqarmitage** [comment]
I didn't notice your patch before. Yes that seems perfectly reasonable since the Linux headers have been modified so that if they are included after the glibc headers, they won't redefine the values (kernel commit 1575c09); this was first included in 4.4.11. So doing the test with net/if.h first followed by linux/if.h will not exhibit the collision problem, whereas the other way round will.

I have merged commit e4dd33f which incorporates your patch.

**Comment by thresheek** [comment]
Many thanks!
